src/include/migraphx/matcher.hpp

```diff
@@ -371,7 +376,20 @@ void find_matches(Mod& mod, instruction_ref ins, Ms&&... ms)
                 std::cout << "Matched by " << get_type_name(m) << std::endl;
                 get_module(mod).debug_print(ins);
             }
+            // If its already invalid dont validate it again
+            bool invalidated = validate and get_module(mod).validate() != get_module(mod).end();
```

Comment:
Why would it continue matching with other matchers if it previously generated an invalid module before from other matcher?

Reply:
This would only find the first match that produced invalid instruction.

Reply:
> This would only find the first match that produced invalid instruction.

Yes thats the matcher that caused the invalid instruction.